src/analysis/convex/cone/proper.lean

```diff
@@ -46,8 +47,89 @@ protected def closure (K : convex_cone ℝ E) : convex_cone ℝ E :=
 
 @[simp, norm_cast] lemma coe_closure (K : convex_cone ℝ E) : (K.closure : set E) = closure K := rfl
 
-protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
-  a ∈ K.closure ↔ a ∈ closure (K : set E) :=
-iff.rfl
+@[simp] protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
+  a ∈ K.closure ↔ a ∈ closure (K : set E) := iff.rfl
 
 end convex_cone
+
+section definitions
+
+/-- A proper cone is a convex cone `K` that is nonempty and closed. Proper cones have the nice
+property that the dual of the dual of a proper cone is itself. This makes them useful for defining
+cone programs and proving duality theorems. -/
+structure proper_cone (E : Type*) [normed_add_comm_group E] [inner_product_space ℝ E]
+  extends convex_cone ℝ E :=
+(nonempty'  : (carrier : set E).nonempty)
+(is_closed' : is_closed (carrier : set E))
+
+end definitions
+
+namespace proper_cone
+
+section inner_product_space
+
+variables {E : Type*} [normed_add_comm_group E] [inner_product_space ℝ E]
+variables {F : Type*} [normed_add_comm_group F] [inner_product_space ℝ F]
+
+instance : has_coe (proper_cone E) (convex_cone ℝ E) := ⟨λ K, K.1⟩
```

Comment:
This coe instance is a bad idea unless you also add a `simp` lemma that picks a canonical spelling.

Reply:
I use this coe frequently in proofs. What would be a good `simp` lemma to clarify this?

Reply:
```
@[simp] lemma to_convex_cone_eq_coe (K : proper_cone E) : K.1 = K := rfl
```

Reply:
I think better to write it in full,
```lean
@[simp] lemma to_convex_cone_eq_coe (K : proper_cone E) : K.to_convex_cone = K := rfl